packages/iguazu-holocron/README.md

```diff
@@ -120,6 +121,176 @@ const loadDataAsProps = ({ store: { dispatch } }) => ({
+import { queryModuleWithData } from 'iguazu-holocron';
+
+const MyModule = ({ MyOtherModule, someData }) => (
+  <>
+    {/* some jsx */}
+    <MyOtherModule someData={someData} />
+  </>
+)
+
+function loadDataAsProps({ store, ownProps }) {
+  return {
+    MyOtherModule: () => store.dispatch(
+      queryModuleWithData('my-module', { someData: ownProps.someData })
+    ),
+  };
+}
+
+export default connectAsync({ loadDataAsProps })(MyModule);
+```
+
+### `isEmpty`
+
+`isEmpty` checks to see if the module queried has been loaded or if the module received in props is
+still the default empty component.
+
+#### Arguments
+
+| name | type | required | value |
+|---|---|---|---|
+| `module` | `Function` | `true` | A module that has been loaded by one of the above Iguazu adapters |
+
+#### Usage
+
+```js
+import { connectAsync } from 'iguazu';
+import { queryModule, isEmpty } from 'iguazu-holocron';
+
+const MyModule = ({ MyOtherModule }) => (
+  <>
+    {/* some jsx */}
+    {isEmpty(MyOtherModule) ? <Spinner /> : <MyOtherModule />}
+  </>
+)
+
+function loadDataAsProps({ store }) {
+  return {
+    MyOtherModule: () => store.dispatch(queryModule('my-module')),
+  };
+}
+
+export default connectAsync({ loadDataAsProps })(MyModule);
+```
+
+### `anyAreEmpty`
+
+`anyAreEmpty` taks `n` arguments, all of which are Holocron modules loaded with one of the above
+Iguazu adapters. It is used to check if any modules in the argument list are still the default empty
+module.
+
+#### Arguments
+
+| name | type | required | value |
+|---|---|---|---|
+| `module` | `Function` | `true` | A module that has been loaded by one of the above Iguazu adapters |
+
+#### Usage
+
+```js
+import { connectAsync } from 'iguazu';
+import { queryModule, anyAreEmpty } from 'iguazu-holocron';
+
+const MyModule = ({ MyOtherModule, ThirdModule }) => (
+  <>
+    {/* some jsx */}
+    { anyAreEmpty(MyOtherModule, ThirdModule) ? 
+      <Spinner /> :
+      <MyOtherModule><ThirdModule/></MyOtherModule>
+    }
+  </>
+)
+
+function loadDataAsProps({ store }) {
+  return {
+    MyOtherModule: () => store.dispatch(queryModule('my-module')),
+    ThirdModule: () => store.dispatch(queryModule('third-module')),
+  };
+}
+
+export default connectAsync({ loadDataAsProps })(MyModule);
+```
+
+### `configureIguazuSSR`
+
+`configureIguazuSSR` is a helper for loading data on the server via Iguazu. It does not need to be 
+called directly, but needs to be attached to the module along with the `loadDataAsProps`.
+
+#### Usage
+
+```js
+import { connectAsync } from 'iguazu';
+import { queryModule } from 'iguazu-holocron';
+
+const MyModule = ({ data }) => <p>{`Data: ${data}`}</p>)
+
+function loadDataAsProps({ store }) {
+  return {
+    data: () => store.dispatch(queryData()),
+  };
+}
+
+MyModule.loadDataAsProps = loadDataAsProps;
+
+if (!global.BROWSER) {
+  MyModule.loadModuleData = configureIguazuSSR;
+}
+
+export default connectAsync({ loadDataAsProps })(MyModule);
+```
+
```

Comment:
Should have `More Information` section for referencing `loadModuleData` Module Lifecycle Hook. Also we dont need `if (!global.BROWSER)` for this function as it is universal. The if-check is for `appConfig` only.